xray_vision/backend/mpl/cross_section_2d.py

```diff
@@ -357,15 +390,61 @@ def __init__(self, fig, cmap=None, norm=None,
 
         # set up all the other axes
         # (set up the horizontal and vertical cuts)
-        self._ax_h = divider.append_axes('top', .5, pad=0.1,
-                                         sharex=self._im_ax)
-        self._ax_h.yaxis.set_major_locator(LinearLocator(numticks=2))
-        self._ax_v = divider.append_axes('left', .5, pad=0.1,
-                                         sharey=self._im_ax)
+
+        # Orientation types and changing plots based on arguments
+        enum_left_right = {'left': True, 'right': False}
+
+        kwargs_vert = {'position': self.vert_loc, 'size': .5, 'pad': 0.1}
+        kwargs_horiz = {'position': self.horiz_loc, 'size': .5}
+        if self.extent_labels is None:
+            kwargs_vert['sharey'] = self._im_ax
```

Comment:
- Initially there was no reason to get rid of the shared axes, but when they are shared the extent labels appear on the image rather than the vertical and horizontal cuts.

Reply:
Interesting. The code seems to be right since it applies the labels/ticks to the correct axes, `self._ax_h` and `self._ax_v`. Can it be a matplotlib bug? @tacaswell, thoughts?